crates/sdk/src/masp/utilities.rs

```diff
@@ -508,17 +519,23 @@ impl MaspClient for IndexerMaspClient {
             }
         }
 
-        let mut stream_of_fetches = stream::iter(fetches)
-            .buffer_unordered(self.shared.max_concurrent_fetches);
+        let mut stream_of_fetches =
+            stream::iter(fetches).buffered(self.shared.max_concurrent_fetches);
```

Comment:
This change won't matter, because we're using a `BTreeMap` to sort the fetched txs based on their `IndexedTx` key. Until a fix on the node software is available, I think we should change the inner type of the `Fetched` struct to a `Vec`, so that we preserve the order of the txs returned by the indexer's webserver.

Reply:
It doesn't create extra problems, it's just that the order returned by the client (in your PR) is not respected, because all fetched masp txs end up in the `Fetched` cache, which sorts txs based on the returned `IndexedTx`.